web/views.py

```
@@ -5773,8 +5786,14 @@ def award_badge(request):
         return JsonResponse({"success": False, "message": "Student not found"}, status=404)
     except Course.DoesNotExist:
         return JsonResponse({"success": False, "message": "Course not found"}, status=404)
-    except Exception:
-        return JsonResponse({"success": False, "message": "Error: award_badge"}, status=500)
+    except (User.DoesNotExist, Course.DoesNotExist) as e:
```

Ruff (0.8.2)

try-except block with duplicate exception `User.DoesNotExist`

(B025)